From ac9c9f580722fcdab4c4222adf6a07c150389b41 Mon Sep 17 00:00:00 2001
From: "Phuong Viet. Nguyen" <phuong.nguyen.xw@rvc.renesas.com>
Date: Tue, 6 Mar 2018 20:41:00 +0700
Subject: [PATCH 255/909] usb: renesas_usbhs: Add support for R-Car E3

commit 4d31b267c1793f16737ce94d105263ef52046980 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch adds support for R-Car E3. This SoC needs to release
the PLL reset by the UGCTRL register. Same as R-Car D3 SoCs.

Signed-off-by: Phuong Nguyen <phuong.nguyen.xw@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/renesas_usbhs/common.c | 4 ++++
 drivers/usb/renesas_usbhs/rcar3.c  | 6 +++---
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/renesas_usbhs/common.c b/drivers/usb/renesas_usbhs/common.c
index 573c9475bcfa..01864730b167 100644
--- a/drivers/usb/renesas_usbhs/common.c
+++ b/drivers/usb/renesas_usbhs/common.c
@@ -485,6 +485,10 @@ static const struct of_device_id usbhs_of_match[] = {
 		.compatible = "renesas,usbhs-r8a77995",
 		.data = (void *)USBHS_TYPE_RCAR_GEN3_WITH_PLL,
 	},
+	{
+		.compatible = "renesas,usbhs-r8a77990",
+		.data = (void *)USBHS_TYPE_RCAR_GEN3_WITH_PLL,
+	},
 	{
 		.compatible = "renesas,usbhs-r8a77965",
 		.data = (void *)USBHS_TYPE_RCAR_GEN3,
diff --git a/drivers/usb/renesas_usbhs/rcar3.c b/drivers/usb/renesas_usbhs/rcar3.c
index d0ea4ff89622..8e0067576aca 100644
--- a/drivers/usb/renesas_usbhs/rcar3.c
+++ b/drivers/usb/renesas_usbhs/rcar3.c
@@ -18,7 +18,7 @@
 /* Low Power Status register (LPSTS) */
 #define LPSTS_SUSPM	0x4000
 
-/* R-Car D3 only: USB General control register (UGCTRL) */
+/* R-Car D3/E3 only: USB General control register (UGCTRL) */
 #define UGCTRL_PLLRESET		0x00000001
 #define UGCTRL_CONNECT		0x00000004
 
@@ -32,7 +32,7 @@
 #define UGCTRL2_USB0SEL_OTG	0x00000030
 #define UGCTRL2_VBUSSEL		0x00000400
 
-/* R-Car D3 only: USB General status register (UGSTS) */
+/* R-Car D3/E3 only: USB General status register (UGSTS) */
 #define UGSTS_LOCK		0x00000100
 
 static void usbhs_write32(struct usbhs_priv *priv, u32 reg, u32 data)
@@ -76,7 +76,7 @@ static int usbhs_rcar3_power_ctrl(struct platform_device *pdev,
 	return 0;
 }
 
-/* R-Car D3 needs to release UGCTRL.PLLRESET */
+/* R-Car D3/E3 needs to release UGCTRL.PLLRESET */
 static int usbhs_rcar3_power_and_pll_ctrl(struct platform_device *pdev,
 					  void __iomem *base, int enable)
 {
-- 
2.17.1

